components/eamxx/scripts/test_all_scream.py

```diff
@@ -886,17 +887,14 @@ def generate_baselines(self, test, commit):
                     print(f"WARNING: Failed to create baselines:\n{err}")
                     success = False
 
-        finally:
-            # Clean up the directory, by removing everything but the 'data' subfolder. This must
-            # happen unconditionally or else subsequent runs could be corrupted
-            run_cmd_no_fail(r"find -maxdepth 1 -not -name data ! -path . -exec rm -rf {} \;",
-                            from_dir=test_dir, verbose=True, dry_run=self._dry_run)
-
         if success:
             # Store the sha used for baselines generation
             self.set_baseline_file_sha(test, commit)
             test.missing_baselines = False
 
+            # Clean up the directory by removing everything
+            shutil.rmtree(test_dir)
```

Comment:
We may need to add one of these also _before_ building baselines. I'm thinking of this scenario:

1. run tas: baseline build fails, so we don't nuke the dir
2. change something, run tas again: baseline dir is not clean, so we may not get a clean build